lib/oauthutil/oauthutil.go

```diff
@@ -849,10 +846,14 @@ func (s *authServer) Init() error {
 	mux.HandleFunc("/", s.handleAuth)
 
 	var err error
-	s.listener, err = net.Listen("tcp", s.bindAddress)
-	if err != nil {
-		return err
+	if s.listener, err = net.Listen("tcp", "[::1]:"+bindPort); err != nil {
```

Comment:
@ncw do you like this approach better? I'll still need to clean up some of the duplicate state like authUrl, redirectUrl, oauthConfig etc.

Reply:
Yes.

Does `[::1]` include `127.0.0.1`? If not then we need to bind to both if possible which is a little awkward!

Reply:
I'm still researching options to listen to both tcp4 and tcp6.   From net.Listen docs we can either (a) listen on a specific address or (b) listen on all multicast/unicast (not safe)

Here are the options I'm aware of to listen to ::1 & 127.0.0.1
 
1. Implement a MultiListener interface in GO (about 50 lines -- a common approach)
2. spawn (2) net.LIsten & http.Serve objects with the identical handler routine
3. TBD -- find the appropriate SetSockOpt / SockAddr struct config to listen on ::1 & 127.0.0.1 but not external interfaces

 I'm continuing research on (3) by reviewing net.go implementation and the C structs man pages.